torch/_dynamo/guards.py

```diff
@@ -1208,26 +1822,34 @@ def add_code_part(code_part, guard, log_only=False):
             **SYMPY_INTERP,
             **CLOSURE_VARS,
         }
+        globals_for_guard_fn = {"G": builder.scope["G"]}
 
-        unique_code_parts = list(unique(code_parts))
-        make_guard_fn_args = ", ".join(closure_vars.keys())
-        guard_body, pycode = build_guard_function(unique_code_parts, make_guard_fn_args)
+        if config.enable_cpp_guard_manager:
+            # Guard manager construction is complete
```

Comment:
Let's add an assert here that `code_parts` is empty (and modify the above not to call add_code_parts) -- this will ensure we didn't miss anything.

Reply:
Ah, very nice! Yes, let me make the change.

Reply:
Done